packages/augur-ui/src/modules/app/components/inner-nav/category-filters.tsx

```diff
@@ -0,0 +1,318 @@
+
+interface CategoryFiltersProps {
+  isSearching: boolean;
+  categoryMetaData: CategoryMetaDataInterface;
+  popularCategories: CategoryInterface[];
+  allOtherCategories: CategoryInterface[];
+}
+
+interface CategoryFiltersState {
+  showAllCategories: boolean;
+  selectedCategories: CategoryInterface[];
+  selectedCategory: CategoryInterface;
+  currentCategories: object;
+}
+
+export default class CategoryFilters extends React.Component<
+  CategoryFiltersProps,
+  CategoryFiltersState
+> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      showAllCategories: false,
+      selectedCategories: [],
+      selectedCategory: null,
+      currentCategories: null,
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.isSearching) {
+      this.setState({
+        showAllCategories: false,
+      });
+    }
+  }
+
+  render() {
+    const hasSelectedCategories = this.state.selectedCategories && this.state.selectedCategories.length > 0;
+    const hasSelectedCategoriesCount = this.state.selectedCategories && this.state.selectedCategories.length;
+
+    const showAllCategoriesButton = (
+      <div
+        className={Styles.AllCategories}
+        onClick={() => this.gotoAllCategories()}
+      >
+        {MenuChevron} All categories
+      </div>
+    );
+
+    return (
+      <div className={Styles.CategoryFilters}>
+        {!hasSelectedCategories && this.renderPopularCategories()}
+        {!hasSelectedCategories && this.props.categoryMetaData && this.renderAllCategories()}
+
+        {hasSelectedCategories && this.props.categoryMetaData && showAllCategoriesButton}
+        {hasSelectedCategories && this.props.categoryMetaData && hasSelectedCategoriesCount > 0 && this.renderSelectedCategories()}
+      </div>
+    );
+  }
+
+  renderPopularCategories() {
+    const renderPopular = this.props.popularCategories.map((item, idx) => {
+      if (this.props.isSearching) {
+        // No meta data yet
+        return (
+          <div key={idx}>
+            <CategoryRow
+              category={item.category}
+              loading={true} />;
+          </div>
+        );
+      }
+
+      return (
+        <div key={idx}>
+          <CategoryRow
+            category={item.category}
+            count={item.count}
+            hasChildren={item.count > 0}
+            handleClick={() => this.getChildrenCategories(item.category)}
+          />
+        </div>
+      );
+    });
+
+    return (
+      <div className={Styles.CategoriesGroup}>
+        <span>Popular Categories</span>
+        {renderPopular}
+      </div>
+    );
+  }
+
+  renderSelectedCategories() {
+    return (
+      <div className={Styles.SelectedCategories}>
+        {this.state.selectedCategories
+          .filter(categories => categories !== this.state.selectedCategory)
+          .map((category, idx) => {
+            return (
+              <div
+                key={idx}
+                className={Styles.SelectedCategories_back}
```

Comment:
don't think this needs its own classname, and also thought we are moving away from underscore classnames

Reply:
👍 good call I will remove the class name completely